### DISCUSSION TITLE
Automating Video Trimming and Cloud Storage with Real-Time Streaming

### DISCUSSION THREAD

**Discussion by davis168** [ant-media/Ant-Media-Server#6667] [Q&A]
I would like to access real-time video streaming, trim the video into short clips through coding, and save the video files to DigitalOcean storage. How should I do it?

**Comment by lastpeony** [comment]
Enable hls streaming
get m3u8 file
parse m3u8 file and extract ts files in desired interval
merge ts files with ffmpeg to mp4
upload mp4 to digital ocean

this is not currently supported out of the box by ant media server but you can do it on your side by writing a side program or ams plugin.

We are currently working on a similar feature and developing it as a plugin.

**Comment by davis168** [comment]
Hi Last Peony,

Thank you for the information. Since this feature isn't currently supported
out of the box by Ant Media Server, could you recommend any tools or
libraries to help with this tasks?

I understand you're working on a plugin for a similar feature, but in the
meantime, any suggestions or recommendations for these tasks would be
greatly appreciated.

Best regards,
Davis Boeng

**Comment by lastpeony** [comment]
ffmpeg obviously to merge ts files to mp4
and a program written by you to get and parse the m3u8. this can be in any language java, python, nodejs etc

each of those languages has various libraries to parse m3u8 easily. just google m3u8 parser java, nodejs etc
after extracting ts file paths you will feed it to a seperate ffmpeg process. then ffmpeg will merge the ts files into an mp4.
